src/data/sym/basic.lean

```diff
@@ -150,4 +155,105 @@ instance inhabited_sym' [inhabited α] (n : ℕ) : inhabited (sym' α n) :=
+  subst h,
+  refl,
+end
+
+instance unique_zero : unique (sym α 0) :=
+⟨⟨nil⟩, eq_nil_of_card_zero⟩
+
+/-- `repeat a n` is the sym containing only `a` with multiplicity `n`. -/
+def repeat (a : α) (n : ℕ) : sym α n := ⟨multiset.repeat a n, multiset.card_repeat _ _⟩
+
+lemma repeat_succ {a : α} {n : ℕ} : repeat a n.succ = a :: repeat a n := rfl
+
+lemma exists_eq_cons_of_succ (s : sym α n.succ) : ∃ (a : α) (s' : sym α n), s = a :: s' :=
+begin
+  classical,
+  obtain ⟨m, h⟩ := s,
+  have : 0 < multiset.card m,
+  { rw h,
+    exact nat.succ_pos n, },
+  rw multiset.card_pos_iff_exists_mem at this,
+  obtain ⟨a, ha⟩ := this,
+  use [a, m.erase a],
+  { rw [multiset.card_erase_of_mem ha, h],
+    refl, },
+  { rw cons,
+    congr,
+    rw multiset.cons_erase ha, },
+end
+
+lemma eq_repeat_of_subsingleton [subsingleton α] (a : α) {n : ℕ} (s : sym α n) : s = repeat a n :=
+begin
+  induction n with n ih,
+  { rw eq_nil_of_card_zero s,
+    refl, },
+  { obtain ⟨b, s', rfl⟩ := exists_eq_cons_of_succ s,
+    rw [repeat_succ, ih s'],
+    congr, },
+end
+
+instance subsingleton (n : ℕ) [subsingleton α] : subsingleton (sym α n) :=
+⟨begin
+  cases n,
+  { simp, },
+  { intros s s',
+    obtain ⟨b, s, rfl⟩ := exists_eq_cons_of_succ s,
+    rw [eq_repeat_of_subsingleton b s', eq_repeat_of_subsingleton b s, repeat_succ], },
+end⟩
+
+instance is_empty (n : ℕ) [is_empty α] : is_empty (sym α n.succ) :=
+⟨λ s, by { obtain ⟨a, s, rfl⟩ := exists_eq_cons_of_succ s, exact is_empty_elim a }⟩
+
+instance unique (n : ℕ) [unique α] : unique (sym α n) := unique.mk' _
+
+lemma repeat_left_injective (n : ℕ) (h : n ≠ 0) : function.injective (λ x : α, repeat x n) :=
+begin
+  intros a b x,
+  simp only [repeat, subtype.mk.inj_eq] at x,
+  exact (multiset.repeat_left_inj a b n h).mp x,
+end
+
+lemma repeat_left_inj (a b : α) (n : ℕ) (h : n ≠ 0) : repeat a n = repeat b n ↔ a = b :=
+(repeat_left_injective n h).eq_iff
+
+instance nontrivial (n : ℕ) [nontrivial α] : nontrivial (sym α (n + 1)) :=
+(repeat_left_injective n.succ n.succ_ne_zero).nontrivial
+
+def map {α β : Type*} {n : ℕ} (f : α → β) (x : sym α n) : sym β n :=
+⟨x.val.map f, by simpa [multiset.card_map] using x.property⟩
+
+@[simp] lemma mem_map {α β : Type*} {n : ℕ} {f : α → β} {b : β} {l : sym α n} :
+  b ∈ sym.map f l ↔ ∃ a, a ∈ l ∧ f a = b := multiset.mem_map
+
+@[simp] lemma map_id {α : Type*} {n : ℕ} (s : sym α n) : sym.map id s = s :=
+by simp [sym.map, subtype.mk.inj_eq]
+
+@[simp] lemma map_map {α β γ : Type*} {n : ℕ} (g : β → γ) (f : α → β) (s : sym α n) :
+  sym.map g (sym.map f s) = sym.map (g ∘ f) s :=
+by simp [sym.map, subtype.mk.inj_eq]
+
+@[simp] lemma map_zero {α β : Type*} (f : α → β) :
+  sym.map f (0 : sym α 0) = (0 : sym β 0) := rfl
+
+@[simp] lemma map_cons {α β : Type*} {n : ℕ} (f : α → β) (a : α) (s : sym α n) :
+  (a :: s).map f = (f a) :: s.map f :=
+by { cases s, simp [map, cons] }
+
+/-- If `α` is equivalent to `β`, then `sym α n` is equivalent to `sym β n`. -/
```

Comment:
Maybe this would be more informative?
```suggestion
/-- Mapping an equivalence `α ≃ β` using `sym.map` gives an equivalence between `sym α n` and `sym β n`. -/
```